staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -930,6 +930,8 @@ func (e *Store) Delete(ctx context.Context, name string, deleteValidation rest.V
 	// TODO: remove the check, because we support no-op updates now.
 	if graceful || pendingFinalizers || shouldUpdateFinalizers {
 		err, ignoreNotFound, deleteImmediately, out, lastExisting = e.updateForGracefulDeletionAndFinalizers(ctx, name, key, options, preconditions, deleteValidation, obj)
+		// Clear out preconditions.ResourceVersion because now it will be invalid
+		preconditions.ResourceVersion = nil
```

Comment:
why wouldn't we update the precondition resource version here?

Reply:
also, this needs to be accompanied by tests demonstrating the issue and why this is the correct fix

Reply:
@liggitt I updated to use the new resource version and added a test as requested.  Without the added code, the new test fails (I have verified locally).

I feel like we should just nil-out the precondition.ResourceVersion as I did in the previous patch because the object will already have a deletion timestamp.  I don't really mind either way, let me know which you prefer.